tests/ord-aggregator/tests/handler_test.go

```diff
@@ -72,205 +77,251 @@ const (
 	testTimeoutAdditionalBuffer = 5 * time.Minute
 )
 
-func TestORDAggregator(t *testing.T) {
-	appInput := fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedSystemInstanceName, expectedSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL)
-	secondAppInput := fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedSecondSystemInstanceName, expectedSecondSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL)
-	thirdAppInput := fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedThirdSystemInstanceName, expectedThirdSystemInstanceDescription, testConfig.ExternalServicesMockAbsoluteURL)
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 
-	systemInstancesMap := make(map[string]string)
-	systemInstancesMap[expectedSystemInstanceName] = expectedSystemInstanceDescription
-	systemInstancesMap[expectedSecondSystemInstanceName] = expectedSecondSystemInstanceDescription
-	systemInstancesMap[expectedThirdSystemInstanceName] = expectedThirdSystemInstanceDescription
+type ordConfigSecurity struct {
+	Enabled  bool   `json:"enabled"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
 
-	eventsMap := make(map[string]string)
-	eventsMap[firstEventTitle] = firstEventDescription
-	eventsMap[secondEventTitle] = secondEventDescription
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
 
-	bundlesMap := make(map[string]string)
-	bundlesMap[expectedBundleTitle] = expectedBundleDescription
-	bundlesMap[secondExpectedBundleTitle] = secondExpectedBundleDescription
+func randString(n int) string {
+	buffer := make([]rune, n)
+	for i := range buffer {
+		buffer[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(buffer)
+}
 
-	bundlesAPIsNumberMap := make(map[string]int)
-	bundlesAPIsNumberMap[expectedBundleTitle] = expectedNumberOfAPIsInFirstBundle
-	bundlesAPIsNumberMap[secondExpectedBundleTitle] = expectedNumberOfAPIsInSecondBundle
+func TestORDAggregator(t *testing.T) {
+	ordConfigSecurity := &ordConfigSecurity{
+		Enabled:  false,
+		Username: randString(8),
+		Password: randString(8),
+	}
 
-	bundlesAPIsData := make(map[string][]string)
-	bundlesAPIsData[expectedBundleTitle] = []string{firstAPIExpectedTitle}
-	bundlesAPIsData[secondExpectedBundleTitle] = []string{firstAPIExpectedTitle}
+	defer func() { // Cleanup just in case (remove security)
+		ordConfigSecurity.Enabled = false
+		toggleORDConfigSecurity(t, testConfig.ExternalServicesMockBaseURL, ordConfigSecurity)
+	}()
+
+	for _, secured := range []bool{false, true} {
```

Comment:
Not sure if needed, but currently this will create the test for unsecured well-known and secured with basic.

However, based on the implementation, technically we support OAuth secured well-known endpoint as well. 

Should we add an e2e test for it as well?

Another thing to consider here is that ord_aggregator tests a pretty slow (~5 mins) now you make it double. If you address this comment - triple. And with the certificate security - maybe 4 times slower ~20min

Reply:
Yeah, I had a similar concern about re-running everything with the only difference being that the 3 apps are once secured and the other iteration they are not.
The best solution IMO would be for the external services mock to have different servers (ports probably) serving the ord content of each app. This way we can customize only specific ord handlers and not the one and only one that represents all 3 apps. So I'll see how to do that and I imagine having again 3 apps:
- 1 unsecured
- 1 basic
- 1 oauth

Eventually 1 for certs.

Reply:
Maybe we should have 2 unsecured - one with baseURL and one with Absolute.

And yes I think this will solve the issue as the tests will still need 1 aggregator run.

For each authentication mechanism, we can just register another system with another type of webhook.

From external services mock PoV this will be just different handlers.